modules/flippBidAdapter.js

```diff
@@ -0,0 +1,110 @@
+import {isEmpty, parseUrl} from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER } from '../src/mediaTypes.js';
+const NETWORK_ID = 11090;
+const AD_TYPES = [4309, 641];
+const TARGET_NAME = 'inline';
+const BIDDER_CODE = 'flipp';
+const ENDPOINT = 'https://gateflipp.flippback.com/flyer-locator-service/prebid_campaigns';
+const DEFAULT_TTL = 30;
+const DEFAULT_CURRENCY = 'USD';
+
+const generateUUID = () => {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, c => {
+    const r = (Math.random() * 16) | 0;
+    const v = c === 'x' ? r : (r & 0x3) | 0x8;
+    return v.toString(16);
+  });
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.siteId) && !!(bid.params.publisherNameIdentifier);
```

Comment:
Can the bid be `null` here?

Reply:
@jpanduro-blackbird  I think it's guaranteed that bid.params is always an object according to prebid.js docs and other adapter implementations in this repo